go/vt/vttablet/tabletmanager/tm_init.go

```diff
@@ -839,3 +841,59 @@ func (tm *TabletManager) hookExtraEnv() map[string]string {
 		"SHARD":        tablet.Shard,
 	}
 }
+
+// initializeReplication is used to initialize the replication when the tablet starts.
+// It returns the current primary tablet for use externally
+func (tm *TabletManager) initializeReplication(ctx context.Context, tabletType topodatapb.TabletType) (primary *topo.TabletInfo, err error) {
+	// If active reparents are disabled, we do not touch replication.
+	// There is nothing to do
+	if *mysqlctl.DisableActiveReparents {
+		return nil, nil
+	}
+
+	// Read the shard to find the current primary, and its location.
+	tablet := tm.Tablet()
+	si, err := tm.TopoServer.GetShard(ctx, tablet.Keyspace, tablet.Shard)
+	if err != nil {
+		return nil, vterrors.Wrap(err, "can't read shard")
+	}
+	if si.PrimaryAlias == nil {
+		// There's no primary. This is fine, since there might be no primary currently
+		log.Warningf("Cannot start replication during initialization: shard %v/%v has no primary.", tablet.Keyspace, tablet.Shard)
+		return nil, nil
+	}
+	if topoproto.TabletAliasEqual(si.PrimaryAlias, tablet.Alias) {
+		// We used to be the primary before we got restarted,
+		// and no other primary has been elected in the meantime.
+		// There isn't anything to do here either.
+		log.Warningf("Cannot start replication during initialization: primary in shard record still points to this tablet.")
+		return nil, nil
+	}
+	currentPrimary, err := tm.TopoServer.GetTablet(ctx, si.PrimaryAlias)
+	if err != nil {
+		return nil, vterrors.Wrapf(err, "Cannot read primary tablet %v", si.PrimaryAlias)
+	}
+
+	durabilityName, err := tm.TopoServer.GetKeyspaceDurability(ctx, tablet.Keyspace)
+	if err != nil {
+		return nil, vterrors.Wrapf(err, "Cannot read keyspace durability %v", tablet.Keyspace)
```

Comment:
`durability policy` to be more clear and consistent.

Reply:
Fixed